src/popup/components/TransactionTagList.vue

```diff
@@ -0,0 +1,157 @@
+
+export default defineComponent({
+  components: { TransactionTag },
+  props: {
+    tx: { type: Object as PropType<ITx>, default: null },
+    isIncomplete: Boolean,
+    isPending: Boolean,
+    isClaim: Boolean,
+    dense: Boolean,
+  },
+  setup(props, { root }) {
+    const { tx } = props;
+
+    const {
+      txType,
+      isAllowance,
+      isDex,
+    } = useTransactionTx({ store: root.$store, tx });
+
+    const account = useGetter<IAccount>('account');
+    const activeNetwork = useGetter<INetwork>('activeNetwork');
+    const availableTokens = useState<ITokenList>('fungibleTokens', 'availableTokens');
+    const getTxDirection = useGetter('getTxDirection');
+
+    const labels = computed<(string | TranslateResult)[]>(() => {
+      if (!tx) return [];
+
+      const transactionTypes = i18n.t('transaction.type') as Record<TransactionType, TranslateResult>;
+
+      if (txType.value?.startsWith('name')) {
+        return [AENS, transactionTypes[txType.value]];
+      }
+      if (txType.value === SCHEMA.TX_TYPE.spend) {
+        return [
+          i18n.t('transaction.type.spendTx'),
+          getTxDirection.value(tx) === TX_FUNCTIONS.received
+            ? i18n.t('transaction.spendType.in')
+            : i18n.t('transaction.spendType.out'),
+        ];
+      }
+      if (isAllowance.value) {
+        return [i18n.t('transaction.dexType.allow_token')];
+      }
+      if (isDex.value) {
+        return [
+          DEX,
+          FUNCTION_TYPE_DEX.pool.includes(tx.function as TxFunctionRaw)
+            ? i18n.t('transaction.dexType.pool')
+            : i18n.t('transaction.dexType.swap'),
+        ];
+      }
+      if (
+        (
+          tx.contractId
+              && (
+                activeNetwork.value.tipContractV1 === tx.contractId
+                  || activeNetwork.value.tipContractV2 === tx.contractId
+              )
+              && (
+                tx.function === TX_FUNCTIONS.tip
+                  || tx.function === TX_FUNCTIONS.retip
+              )
+        )
+          || props.isClaim
+      ) {
+        return [
+          i18n.t('pages.token-details.tip'),
+          props.isClaim
+            ? i18n.t('transaction.spendType.in')
+            : i18n.t('transaction.spendType.out'),
+        ];
+      }
+      if (
+        txType.value === SCHEMA.TX_TYPE.contractCall
+          && availableTokens.value[tx.contractId]
+          && (
+            tx.function === TX_FUNCTIONS.transfer
+              || props.isIncomplete
+          )
+      ) {
+        return [
+          i18n.t('transaction.type.spendTx'),
+          tx.callerId === account.value.address
+            ? i18n.t('transaction.spendType.out')
+            : i18n.t('transaction.spendType.in'),
+        ];
+      }
+
+      if (txType.value === TX_TYPE_MDW.PayingForTx) {
+        return [
+          i18n.t('transaction.type.payingForTx'),
+        ];
+      }
+      return props.isPending ? [] : [transactionTypes[txType.value]];
```

Comment:
Before this return please add this additional if: https://github.com/aeternity/superhero-wallet/pull/1795/files#diff-6f2892100eb0e98f7e0a50d24ad5e09400985127894b862678518d394878e403